roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -55,59 +65,85 @@ def required_images(self, task_vars):
         deployment_type = get_var(task_vars, "deployment_type")
         image_base_name = self.docker_image_base[deployment_type]
 
-        # openshift_release = get_var(task_vars, "openshift_release")
-        # openshift_image_tag = get_var(task_vars, "openshift_image_tag")
-        openshift_release = "v3.4"
-        openshift_image_tag = "3.4.1.7"
+        openshift_release = get_var(task_vars, "openshift_release")
+        openshift_image_tag = get_var(task_vars, "openshift_image_tag")
 
         is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
 
         if is_containerized:
-            docker_or_rpm_images = set(self.containerized_docker_images(image_base_name, openshift_release))
+            images = set(self.containerized_docker_images(image_base_name, openshift_release))
         else:
-            docker_or_rpm_images = set(self.rpm_docker_images(image_base_name, openshift_release))
+            images = set(self.rpm_docker_images(image_base_name, openshift_release))
 
         # append images with qualified image tags to our list of required images.
         # these are images with a (v0.0.0.0) tag, rather than a standard release
         # format tag (v0.0). We want to check this set in both containerized and
         # non-containerized installations.
-        docker_or_rpm_images.update(self.qualified_docker_images(self.image_from_base_name(image_base_name),
-                                                                 "v" + openshift_image_tag))
+        images.update(
+            self.qualified_docker_images(self.image_from_base_name(image_base_name), "v" + openshift_image_tag)
+        )
 
-        return docker_or_rpm_images
+        return images
 
     def local_images(self, required_images, task_vars):
-        """ Return all found local docker images """
-        found = set()
-        for image in required_images:
-            cmd = self.module_executor("docker_image_facts", {"name": image}, task_vars)
-            if cmd.get("images", None):
-                found.add(image)
+        """Filter a list of images and return those available locally."""
+        return [
+            image for image in required_images
+            if self.is_image_local(image, task_vars)
+        ]
 
-        return found
+    def is_image_local(self, image, task_vars):
+        result = self.module_executor("docker_image_facts", {"name": image}, task_vars)
+        if result.get("failed", False):
+            return False
+
+        return bool(result.get("images", []))
 
     def known_docker_registries(self, task_vars):
-        docker_info = self.module_executor("docker_info", {}, task_vars).get("info", "")
+        result = self.module_executor("docker_info", {}, task_vars)
+
+        # do test on module exception - see if this line is useful
```

Comment:
Although, in the case of a module error due to the docker daemon being down, the entire check would fail before getting to this point, due to `docker_image` failing to connect during the `self.update_skopeo_image` step